### PR TITLE
Feature/goldstone csv

### PR DESCRIPTION
**Pull request by talister**

### CODE REVIEW

File: neoexchange/astrometrics/tests/test_sources_subs.py [hunk 1]
```diff
@@ -415,7 +416,36 @@ def test_page_down(self):
...
+    def setUp(self):
+        self.test_file = os.path.join('astrometrics', 'tests', 'test_goldstone_page.csv')
+
+    def test_basic(self):
+        expected_length = 7
+        expected_columns = ['number', 'name', 'start (UT)', 'end (UT)', 'OCC', 'Updated 2021 Dec 17']
+
+        table = fetch_goldstone_csv(self.test_file)
+
+        self.assertEqual(expected_length, len(table))
```
Comment (jchate6) [inline neoexchange/astrometrics/tests/test_sources_subs.py:430]: Shouldn't the expected length be 9? I don't think do any cuts yet.

Comment (jchate6) [reply neoexchange/astrometrics/tests/test_sources_subs.py:430]: also, this test fails for me.

File: neoexchange/astrometrics/sources_subs.py [hunk 2]
```diff
@@ -1285,6 +1287,42 @@ def fetch_goldstone_targets(page=None, dbg=False):
...
+                target = row['name']
+            start_date = datetime.strptime(row['start (UT)'], '%m/%d/%Y')
+            if calendar_format is True:
+                end_date = datetime.strptime(row['end (UT)'], '%m/%d/%Y')
+                end_date += timedelta(seconds=86399)
+                target = {'target' : target,
+                          'windows' : [{'start' : start_date.strftime("%Y-%m-%dT%H:%M:%S"),
+                                        'end' : end_date.strftime("%Y-%m-%dT%H:%M:%S")}]
+                             }
+            if start_date.year == current_year:
```
Comment (jchate6) [inline neoexchange/astrometrics/sources_subs.py:1322]: I know we don't tend to plan very far ahead, but would it make sense to think about uploading the next year's targets in December rather than waiting until January?

File: neoexchange/astrometrics/tests/test_sources_subs.py [hunk 3]
```diff
@@ -619,6 +650,69 @@ def test_target_with_ampersand4(self):
...
+                             {'target': '2018 CW2',
+                              'windows': [{'start': '2022-02-16T00:00:00', 'end': '2022-02-21T23:59:59'}]},
+                             {'target': '2010 XC15',
+                              'windows': [{'start': '2022-12-24T00:00:00', 'end': '2023-01-06T23:59:59'}]},
+                              ]
+
+
+        targets = fetch_goldstone_targets(self.test_csv_file, calendar_format=True)
+
+        self.assertEqual(len(expected_targets), len(targets))
```
Comment (jchate6) [inline neoexchange/astrometrics/tests/test_sources_subs.py:713]: This test seems unnecessary given the next one...

File: neoexchange/astrometrics/tests/test_sources_subs.py [hunk 4]
```diff
@@ -415,7 +416,36 @@ def test_page_down(self):
...
+        expected_length = None
+
+        table = fetch_goldstone_csv('/foo/bar')
+
+        self.assertEqual(expected_length, table)
+
+    def test_missing_url(self):
+        expected_length = None
+
+        table = fetch_goldstone_csv('https://foo.bar.com/nothere.csv')
```
Comment (jchate6) [inline neoexchange/astrometrics/tests/test_sources_subs.py:443]: Excited for this domain to be claimed and populated in the future. :P

**Comment by jchate6** [comment]
For the record, the failing test at `neox.tests.test_plot_lc.LighCurvePlotTest` will be fixed by merging `main`.

File: neoexchange/astrometrics/tests/test_sources_subs.py (hunk 4 above)
Comment (talister) [reply neoexchange/astrometrics/tests/test_sources_subs.py:443]: It needed to be a properly formatted domain otherwise it failed differently inside _astropy_s `request`-wrapped internals... Slightly surprised someone hasn't grabbed it already...

File: neoexchange/astrometrics/tests/test_sources_subs.py (hunk 3 above)
Comment (talister) [reply neoexchange/astrometrics/tests/test_sources_subs.py:713]: It's cleaner to see a basic "is there the correct number of entries" test first; the list of JSON dictionaries gets a bit long and hard to parse in the test output when they differ

File: neoexchange/astrometrics/tests/test_sources_subs.py (hunk 1 above)
Comment (talister) [reply neoexchange/astrometrics/tests/test_sources_subs.py:430]: Fixed in 2d71c1a556cf0b2ac79a6defcce29b31aff71bc0

File: neoexchange/astrometrics/sources_subs.py (hunk 2 above)
Comment (talister) [reply neoexchange/astrometrics/sources_subs.py:1322]: Maybe within 2 weeks of the end of December, we allow the next year's targets as valid ?

Comment (jchate6) [reply neoexchange/astrometrics/sources_subs.py:1322]: might make some sense, though traditionally, those weeks are great for work.

File: neoexchange/astrometrics/tests/test_sources_subs.py (hunk 3 above)
Comment (jchate6) [reply neoexchange/astrometrics/tests/test_sources_subs.py:713]: fair enough.

**Review by jchate6 — APPROVED** [review]
